cpp/src/arrow/buffer.h

```diff
@@ -173,7 +173,7 @@ class ARROW_EXPORT ResizableBuffer : public MutableBuffer {
 /// A Buffer whose lifetime is tied to a particular MemoryPool
 class ARROW_EXPORT PoolBuffer : public ResizableBuffer {
  public:
-  explicit PoolBuffer(MemoryPool* pool = nullptr);
+  PoolBuffer(MemoryPool* pool = nullptr);
```

Comment:
Why did you remove this?

Reply:
This will fail the cpplint checks so I wanted to make sure that my changes have the desired effect of failing the builds fast